src/main/java/org/logstash/dissect/Delimiter.java

```diff
@@ -0,0 +1,42 @@
+package org.logstash.dissect;
+
+import org.logstash.dissect.search.DelimiterSearchStrategy;
+
+public final class Delimiter {
+    private final byte[] needle;
+    private final String delimiter;
+    private final int size;
+    private DelimiterSearchStrategy strategy;
+
+    public Delimiter(String delim) {
+        delimiter = delim;
+        needle = delim.getBytes();
+        size = needle.length;
+    }
+
+    public Delimiter addStrategy(DelimiterSearchStrategy strategy) {
```

Comment:
Do you really need to change this strategy over the lifetime of a `Delimiter` instance? If not you can pass it in the constructor and declare the corresponding field as `final` thus making `Delimiter` immutable (always good :)).